safehttp/mux.go

```diff
@@ -97,14 +97,38 @@ func NewServeMux(d Dispatcher, domains ...string) *ServeMux {
 
 // Handle registers a handler for the given pattern and method. If another
 // handler is already registered for the same pattern and method, Handle panics.
-func (m *ServeMux) Handle(pattern string, method string, h Handler) {
+//
+// Configs can be optionally passed in order to modify the behavior of the
+// interceptors on a registered handler. Passing a Config whose corresponding
+// Interceptor was not installed will produce no effect.
+func (m *ServeMux) Handle(pattern string, method string, h Handler, configs ...Config) {
+	interceps := map[string]Interceptor{}
+	for k, i := range m.interceps {
+		interceps[k] = i
+	}
+	var changed bool
+	for _, c := range configs {
+		for k, i := range interceps {
```

Comment:
I assume that there will typically be more interceptors than configs, so it might be better to swap these two loops around: iterate over interceptors only once, and then apply all the configs.

Reply:
Why is that better? Is it because iterating over a map is more expensive than iterating over a slice and therefore it's better to iterate over the slice many times and over the map just once?

Reply:
Yes, that’s one aspect. Another one I had in mind is memory locality: it’s better to do a few changes at once to each interceptor, rather than go through all the interceptors over and over again.

Probably doesn’t matter here at all though, so feel free to ignore me.

(I had in mind that when working with images, it’s better to iterate over Y in the outer loop, and X in the inner loop, for memory locality.)